src/js/view/controls/settings-menu.js

```diff
@@ -54,6 +51,12 @@ export function createSettingsMenu(controlbar, onVisibility, localization) {
     return settingsMenu;
 }
 
+function showSettingsMenuIcon(settingsMenu, controlbar) {
+    // Show or hide settings menu icon dependant on amount of submenus
+    const submenuNames = settingsMenu.getSubmenuNames();
+    const toggleIcon = submenuNames.indexOf('quality') > -1 || submenuNames.indexOf('playbackRates') > -1;
+    controlbar.elements.settingsButton.toggle(toggleIcon || submenuNames.length > 1);
```

Comment:
You don't need `some` or `indexOf` to run at all if you check length first. You should always short circuit expensive operations with cheaper ones like this, and encapsulate all `toggleIcon` logic in one place:

```js
const toggleIcon = submenuNames.length > 1 || submenuNames.some(name => (name === 'quality' || name === 'playbackRates'));
```